aptos-move/framework/aptos-framework/sources/configs/Stake.move

```diff
@@ -668,6 +743,22 @@ module AptosFramework::Stake {
             abort Errors::invalid_argument(EWITHDRAW_NOT_ALLOWED)
         };
 
+        // Update the amount of stake being withdrawn from the active validator set's total voting power.
+        // We only do this if the validator is active.
+        // If the validator is pending_active, it's not possible to unlock immediately if we assume that the min lockup
```

Comment:
I think it's better to verify than assume, when we update min lockup or epoch duration we should check the invariant if we're going down this path. or explicitly assert this pool is not in pending_active vector

Reply:
It's really tricky to check epoch length in Stake as Block already depends on Stake (via Reconfiguration). So if Stake also calls Block to get the epoch length, there'd be a circular dependency. We can technically move the epoch length out of Block into another module, but that sounds like adding a lot more code.

Reply:
we can explicitly prevent withdraw if it's in pending_active then? I'm always skeptical of implicit invariants that may potentially break